extensions/healthie/validation/getPatient.zod.ts

```diff
@@ -0,0 +1,64 @@
+import { type E164Number } from 'libphonenumber-js'
+import {
+  DateTimeSchema,
+  E164PhoneValidationSchema,
+} from '../../../lib/shared/validation'
+import { type ActivityEvent } from '../../../lib/types/ActivityEvent'
+import { type GetUserQuery } from '../gql/sdk'
+
+interface ValidateResult {
+  data: { phoneNumber?: E164Number; dob?: string }
+  events?: ActivityEvent[]
+}
+export const validateGetPatient = (
+  user: GetUserQuery['user']
+): ValidateResult => {
+  const phoneValidationResult = E164PhoneValidationSchema.safeParse(
+    user?.phone_number
+  )
+  const dobValidationResult = DateTimeSchema.safeParse(user?.dob)
+
+  return {
+    data: {
+      dob: dobValidationResult.success ? dobValidationResult.data : undefined,
+      phoneNumber: phoneValidationResult.success
+        ? phoneValidationResult.data
+        : undefined,
+    },
+    events:
```

Comment:
[Q] What happens if Healthie returns the below response

```javascript
{
  user: {
    phone_number: '',
    dob: '',
  }
}

// OR

{
  user: {
    phone_number: undefined,
    dob: undefined,
  }
}
```

Given that both values are empty or undefined, I want to make sure we are not logging an event for it saying that we couldn't parse the values to a phone or date value, respectively. We don't do that for other types either.

So if the value is empty or nil, there should be no reason to try and parse it to a phone or dob because it's just empty/nil and that's fine. We only need to validate when there is an actual value.

Reply:
Ok, I will add that to tests and make sure it works that way

Reply:
@nckhell Here are the test cases. If you want me to put any other specific checks here, just let me know.

![image](https://user-images.githubusercontent.com/29845162/233363986-53e464f3-0ad3-4637-b023-c5c818151cf7.png)